systemvm/debian/opt/cloud/bin/setup/cloud-early-config

```diff
@@ -35,8 +35,15 @@ log_it() {
 }
 
 hypervisor() {
-  local try=$([ -x /usr/sbin/virt-what ] && virt-what | tail -1)
-  [ "$try" != "" ] && echo $try && return 0
+  [ -x /usr/sbin/virt-what ] && local facts=( $(virt-what) )
+  if [ "$facts" != "" ]; then
+    # Xen HVM is recognized as Hyperv when Viridian extensions are enabled
+    if [ "${facts[-1]}" == "xen-domU" ] && [ "${facts[0]}" == "hyperv" ]; then
```

Comment:
We have support for `hyperv`, it may be wrong to return 'xen-hvm' when fact is hyperv? Can you fix this?

Reply:
If I'm not mistaken, we will get ```xen-hvm``` only if the latest fact returned by virt-what is "xen-domU" when the first fact is ```hyperv```. On all other cases, we will get the latest fact.